### ISSUE TITLE
/api/jobs returns 500 on get requests

### ISSUE THREAD

**Issue by uint0** [sphenlee/waterwheel#2]
**Description**
Hitting `/api/jobs` with a GET request results in a internal server error, due `&q.project` being incorrectly accessed in `src/server/api/job.rs:get_by_name`.

**Expected Behaviour**
Should return a 400.

**Steps to reproduce**
1. Run the application
2. `curl localhost:8080/api/jobs -i`

**Logs**
```
$ curl localhost:8080/api/jobs -i
HTTP/1.1 500 Internal Server Error
content-length: 0
date: Thu, 24 Feb 2022 13:29:23 GMT
```
```
[2022-02-25T00:29:23.313+11:00 ERROR highnoon::filter::log]
internal server error
    method: GET
    uri: /api/jobs
    error: missing field `project`
    backtrace: missing field `project`
```

**Comment by sphenlee** [comment]
Thanks for the report. This is actually fixed in highnoon 0.0.7 which I will be upgrading to soon.
Highnoon now returns a Bad Request error if the query string fails to deserialize instead of an Internal Server Error.

That being said I'm not even sure if the `/api/jobs` endpoint is even useful... You can list all jobs in a project at `/api/projects/:id/jobs` so maybe the lookup by name should be moved to this endpoint?

**Comment by uint0** [comment]
Issue was fixed in ad300aefdeae2358b7246d40f6f4c4df1cf73a84 (and sphenlee/highnoon@5203e0f2af5cf21d734c704abd207521157fb521).

I agree that the global job lookup by name doesn't seem to provide much. I'll raise a PR to remove it 😄
